File: 2415. Reverse Odd Levels of Binary Tree.py
class Solution:
    def reverseOddLevels(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        def swapNode(node1, node2, depth):
            if not node1 or not node2: return
            if depth % 2:
                node1.val, node2.val = node2.val, node1.val
            swapNode(node1.left, node2.right, depth + 1)
            swapNode(node1.right, node2.left, depth + 1)
        
        swapNode(root.left, root.right, 1)
        return root

File: 2415. Reverse Odd Levels of Binary Tree.cpp

class Solution {
public:
    void swapNode(TreeNode* p, TreeNode* q, int level)
    {
        if(q == NULL or q == NULL) return;
        
        if(level % 2 == 1) swap(p->val, q->val);

        swapNode(p->left, q->right, level+1);
        swapNode(p->right, q->left, level+1);
    }
    TreeNode* reverseOddLevels(TreeNode* root) 
    {
        swapNode(root->left, root->right,1);
        return root;
    }
};